swift/Arrow/Sources/Arrow/ChunkedArray.swift

```diff
@@ -42,8 +45,32 @@ public class ChunkedArray<T> {
         self.nullCount = nullCount
     }
     
-    subscript(_ index: UInt) -> ArrowArray<T> {
-        return arrays[Int(index)]
+    public subscript(_ index: UInt) -> T? {
+        if arrays.count == 0 {
+            return nil
+        }
+        
+        var localIndex = index
+        var arrayIndex = 0;
+        var len: UInt = arrays[arrayIndex].length
+        while(localIndex > len) {
```

Comment:
Is a space missing here?

```suggestion
        while (localIndex > len) {
```

Reply:
Good catch! I will remove the parenthesis.